core/src/trezor/ui/layouts/tt_v2/__init__.py

```diff
@@ -1053,3 +1080,50 @@ async def request_pin_on_device(
             raise wire.PinCancelled
         assert isinstance(result, str)
         return result
+
+
+class RustProgress:
+    def __init__(
+        self,
+        title: str,
+        description: str | None = None,
+        indeterminate: bool = False,
+    ):
+        self.description = description or ""
+        self.layout: Any = trezorui2.show_progress(
+            title=title.upper(),
+            indeterminate=indeterminate,
+            description=None if description is None else lambda: self.description,
+        )
+        ui.backlight_fade(ui.style.BACKLIGHT_DIM)
+        ui.display.clear()
+        self.layout.attach_timer_fn(self.set_timer)
+        self.layout.paint()
+        ui.backlight_fade(ui.style.BACKLIGHT_NORMAL)
+
+    def set_timer(self, token: int, deadline: int) -> None:
+        raise RuntimeError  # progress layouts should not set timers
+
+    def request_complete_repaint(self) -> None:
```

Comment:
Right, I thought this will be needed for cases when the progress is interleaved with confirmation dialogs, however this only happens in `apps/bitcoin/sign_tx/progress.py` and we just reinitialize whole layout object instead. 3f8c350